instrumentation/redis/lib/opentelemetry/instrumentation/redis/patches/client.rb

```diff
@@ -69,16 +96,31 @@ def parse_commands(commands) # rubocop:disable Metrics/AbcSize
               # If we receive an authentication request command
               # we want to short circuit parsing the commands
               # and return the obfuscated command
-              return 'AUTH ?' if command[0] == :auth
+              if command[0] == :auth
+                ret = "AUTH ?"
+                command_bytes += "AUTH ".bytesize + command[1].bytesize
+                return command_bytes, ret
+              end
 
               if config[:db_statement] == :obfuscate
-                command[0].to_s.upcase + ' ?' * (command.size - 1)
+                cmd = "#{command[0].upcase}"
+                command_bytes += cmd.bytesize
+                cmd += ' ?' * (command.size - 1)
+                command_bytes += " #{command[1..-1].join(' ')}".bytesize
+                cmd
               else
                 command_copy = command.dup
                 command_copy[0] = command_copy[0].to_s.upcase
-                command_copy.join(' ')
+                cmd = command_copy.join(' ')
+                command_bytes += cmd.bytesize
+                cmd
               end
             end.join("\n")
+
+            ret = OpenTelemetry::Common::Utilities.utf8_encode(ret, binary: true)
```

Comment:
This is now in `parse_commands` because  the next line's `count` call will error out if the command is an invalid byte sequence in UTF-8.